src/resources/js/views/rsvp.js

```diff
@@ -129,55 +137,100 @@
-		if (
-			! $.trim( name.val() ).length
-			|| ! $.trim( email.val() ).length
-			|| parseFloat( qty.val() ) < 1
-		) {
-			form.find( '.tribe-block__rsvp__message__error' ).show();
-			return false;
+		var $ticket   = $( this ).closest( '.tribe-block__rsvp__ticket' );
+		var ticket_id = $ticket.data( 'rsvp-id' );
+		var $form     = $ticket.find( 'form' );
+
+		var is_rsvp_valid = validate_submission( $form );
+		var is_meta_valid = true;
+		var has_tickets_plus = !! window.tribe_event_tickets_plus;
+
+		if ( has_tickets_plus ) {
+			is_meta_valid = window.tribe_event_tickets_plus.meta.block_validate_meta( $form );
 		}
 
-		var params = form.serializeArray();
-		params.push( { name: 'action', value: 'rsvp-process' } );
-		params.push( { name: 'ticket_id', value: ticket_id } );
-		tribe_rsvp_loader_start( ticket );
-		$.post(
-			TribeRsvp.ajaxurl,
-			params,
-			function( response ) {
-				// Get the remaining number
-				var remaining = response.data.remaining;
+		// Handle invalid form
+		if ( ! is_rsvp_valid || ! is_meta_valid ) {
+			is_rsvp_valid
+				? $form.find( '.tribe-block__rsvp__message__error' ).hide()
+				: $form.find( '.tribe-block__rsvp__message__error' ).show();
+			has_tickets_plus && is_meta_valid
+				? $form.find( '.tribe-event-tickets-meta-required-message' ).hide()
+				: $form.find( '.tribe-event-tickets-meta-required-message' ).show();
+
+			$( 'html, body').animate({
+				scrollTop: $form.offset().top - 100,
+			}, 300 );
+		} else {
+			// Form is valid, submit form
+			var params = $form.serializeArray();
+			params.push( { name: 'action', value: 'rsvp-process' } );
+			params.push( { name: 'ticket_id', value: ticket_id } );
+
+			tribe_rsvp_loader_start( $ticket );
 
-				// Update the remaining template part
-				ticket.find( '.tribe-block__rsvp__details .tribe-block__rsvp__availability' ).replaceWith( response.data.remaining_html );
+			$.post(
+				TribeRsvp.ajaxurl,
+				params,
+				function( response ) {
+					// Get the remaining number
+					var remaining = response.data.remaining;
+
+					// Update templates
+					$ticket.find( '.tribe-block__rsvp__details .tribe-block__rsvp__availability' ).replaceWith( response.data.remaining_html );
+					$ticket.find( '.tribe-block__rsvp__form' ).html( response.data.html );
 
-				ticket.find( '.tribe-block__rsvp__form' ).html( response.data.html );
+					if ( 0 === remaining ) {
+						// If there are no more RSVPs remaining we update the status section
+						$ticket.find( '.tribe-block__rsvp__status' ).replaceWith( response.data.status_html );
+					}
 
-				if ( 0 === remaining ) {
-					// If there are no more RSVPs remaining we update the status section
-					ticket.find( '.tribe-block__rsvp__status' ).replaceWith( response.data.status_html );
+					tribe_rsvp_loader_end( $ticket );
 				}
+			);
+		}
+	}
 
-				tribe_rsvp_loader_end( ticket );
-			}
-		);
+	/**
+	 * Bind events to elements
+	 */
+	function bind_events() {
+		$( '.tribe-block__rsvp__ticket' )
```

Comment:
save the `.tribe-block__rsvp__ticket` into a variable to avoid doing another search.